### DISCUSSION TITLE
Improving Organic, using Android Studio tools

### DISCUSSION THREAD

**Discussion by chrisdebian** [organicmaps/organicmaps#7068] [Ideas]
Hi, everybody.

I'm really impressed by Organic Maps, and was wondering how I could contribute.

I don't know Java or Kotlin, but am interested in code quality.

I installed Android Studio, and opened the latest APK, from GitHub. I am new to Android Studio, but am intrigued by the code analysis that can be done by Studio. I asked Studio to report any perceived problems, and as you'd expect, it returned lots of results. I suspect most of these are false positives, but as no code is ever perfect, some of the findings could be real issues.

I want to find a repeatable way to work through the reported 'errors', but don't know enough about Studio, to dive straight in. Can anyone help?

At the moment, I need to do the following, does this sound right?

1. Check that Studio is properly configured (how?),
2. Import the APK, or ideally work out how Studio integrates with GitHub, without me breaking anything .
3. Work out the best way to identify code problems, or flag 'quality' issues. Several options exist in the Studio menus.
4. Potentially report any issues, as bugs. I'd ideally be able to fix things myself and then submit my suggestion, for consideration.
5. Do this all again, building my knowledge, hopefully.
6. Automate this.

I'd appreciate any thoughts you may have, in the above workflow, and welcome any improvements to it.

Many thanks,

Chris
chris_debian

**Comment by Jean-BaptisteC** [comment]
Good suggestion

You can find main warnings that can be fixed by executing `./gradlew lint`
This linter generates warnings about deprecations and warnings on resources files.

**Reply by chrisdebian** [reply]
On a Linux box, where would I execute the lint command? The APK is currently stored in ~/Downloads

Thanks,

Chris.

**Comment by chrisdebian** [comment]
Thanks @Jean-BaptisteC , I'm new to all this, so will have to find my way, slowly.

Cheers,

Chris

**Comment by chrisdebian** [comment]
Ok, so i've just had a chance to fire-up Android Studio. As I said before, loads of issues are flagged, but most are probably false positives.

To work out the best way to approach this, can someone comment on this screenshot?

![Screenshot from 2024-01-08 15-15-39](https://github.com/organicmaps/organicmaps/assets/17935223/b2b018c4-8088-41f5-bdc1-da28045550b6)

What is the best way to report this potential security issue? It seems to suggest that potentially incorrect permissions are being used, I think.

Thanks,

Chris.

**Reply by Jean-BaptisteC** [reply]
It's not the right solution to analyze apk, it's better to analyze code from the master branch.
How do you want to improve code, if you don't know Java sometimes Android Studio can give false positives and potentially generate crashes.

**Comment by chrisdebian** [comment]
Thank you
